src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp

```diff
@@ -292,6 +362,20 @@ ReportPlatform::writeReport( const xrt_core::device* /*_pDevice*/,
     _output << std::endl;
   }
 
+  const boost::property_tree::ptree& macs = _pt.get_child("platform.macs", pt_empty);
+  if(!macs.empty()) {
+    _output << std::endl;

```

Comment:
This code can be simplified to something like this:
```
if (!macs.empty()) {
  std::string formattedStr;
  for (auto & km : macs) 
     formattedStr += boost::str(fmtBasic % formattedStr().empty() ? "Mac Address" : "" % km.second.get<std::string>("address", "N/A");}
}
```
And yes, this maybe overly simplified, but I do believe you get the idea on what is happening here.

Reply:
done